src/libs/ReportUtils.ts

```diff
@@ -974,57 +1057,54 @@ function getDefaultWorkspaceAvatar(workspaceName) {
+    const workspaceName = getPolicyName(report, false, allPolicies?.[`${ONYXKEYS.COLLECTION.POLICY}${report?.policyID}`]);
+    return allPolicies?.[`policy${report?.policyID}`]?.avatar ?? getDefaultWorkspaceAvatar(workspaceName);
 }
 
 /**
  * Returns the appropriate icons for the given chat report using the stored personalDetails.
  * The Avatar sources can be URLs or Icon components according to the chat type.
- *
- * @param {Array} participants
- * @param {Object} personalDetails
- * @returns {Array<*>}
  */
-function getIconsForParticipants(participants, personalDetails) {
-    const participantDetails = [];
+function getIconsForParticipants(participants: number[], personalDetails: OnyxCollection<PersonalDetails>): Avatar[] {
+    const participantDetails: ParticipantDetails[] = [];
     const participantsList = participants || [];
 
-    for (let i = 0; i < participantsList.length; i++) {
-        const accountID = participantsList[i];
-        const avatarSource = UserUtils.getAvatar(lodashGet(personalDetails, [accountID, 'avatar'], ''), accountID);
-        const displayNameLogin = lodashGet(personalDetails, [accountID, 'displayName']) || lodashGet(personalDetails, [accountID, 'login'], '');
-        participantDetails.push([accountID, displayNameLogin, avatarSource, lodashGet(personalDetails, [accountID, 'fallBackIcon'])]);
+    for (const accountID of participantsList) {
+        const avatarSource = UserUtils.getAvatar(personalDetails?.[accountID]?.avatar ?? '', accountID);
+        const displayNameLogin = personalDetails?.[accountID]?.displayName ? personalDetails?.[accountID]?.displayName : personalDetails?.[accountID]?.login;
+        participantDetails.push([accountID, displayNameLogin ?? '', avatarSource, personalDetails?.[accountID]?.fallbackIcon ?? '']);
     }
 
-    const sortedParticipantDetails = _.chain(participantDetails)
-        .sort((first, second) => {
-            // First sort by displayName/login
-            const displayNameLoginOrder = first[1].localeCompare(second[1]);
-            if (displayNameLoginOrder !== 0) {
-                return displayNameLoginOrder;
-            }
+    const sortedParticipantDetails = participantDetails.sort((first, second) => {
+        // First sort by displayName/login
+        const displayNameLoginOrder = first[1].localeCompare(second[1]);
+        if (displayNameLoginOrder !== 0) {
+            return displayNameLoginOrder;
+        }
 
-            // Then fallback on accountID as the final sorting criteria.
-            // This will ensure that the order of avatars with same login/displayName
-            // stay consistent across all users and devices
-            return first[0] > second[0];
-        })
-        .value();
+        // Then fallback on accountID as the final sorting criteria.
+        // This will ensure that the order of avatars with same login/displayName
+        // stay consistent across all users and devices
+        return first[0] - second[0];
```

Comment:
Make sure it's correct logic since you changed `>` to `-`

Reply:
There is a test specifically for this function , so I am pretty confident about that change 😄